opendevin/llm/llm.py

```diff
@@ -205,16 +209,48 @@ def wrapper(*args, **kwargs):
                 debug_message += message_separator + message['content']
             llm_prompt_logger.debug(debug_message)
 
-            # call the completion function
-            resp = completion_unwrapped(*args, **kwargs)
+            async def check_stopped():
+                while True:
+                    if (
+                        self.stop_requested_callback is not None
+                        and await self.stop_requested_callback()
+                    ):
+                        return True
+                    await asyncio.sleep(0.1)  # Check every 100ms
+
+            async def run_litellm():
+                return await asyncio.to_thread(
+                    partial(completion_unwrapped, *args, **kwargs)
+                )
+
+            litellm_task = asyncio.create_task(run_litellm())
```

Comment:
Just to note, litellm has async variants of its functions, such as `acompletion` for `completion`, it may be worth taking a look at it

Reply:
Glad you mentioned that! Just switched my code locally and will try it out in a few. Cheers!